src/sqlfluff/testing/rules.py

```diff
@@ -85,12 +85,8 @@ def assert_rule_fail_in_sql(code, sql, configs=None, line_numbers=None):
                     line_numbers, actual_line_numbers
                 )
             )
-    # The query should already have been fixed if possible so just return the raw.
-    if linted.num_violations(fixable=True) > 0:
-        fixed, _ = linted.fix_string()
-        return fixed
-    else:
-        return linted.tree.raw
+    fixed, _ = linted.fix_string()
```

Comment:
This code was overly complex and wrong. For tests using Jinja, we need to call `fix_string()` to get the pre-rendered , fixed code.